psm_lib.py

```
@@ -16,53 +41,335 @@ def calculate_propensity_score(df, treatment_col, covariate_cols):
+            )
+        }
+
+        # Define the computation logic as an inner function
+        def _compute_propensity_score():
+            # === INTEGRATION: Memory Check ===
+            MEMORY_MANAGER.check_and_cleanup()
+
+            logger.info("🔄 Computing propensity scores (Logic Execution)...")
+            
+            # Prepare Data: Drop NAs in relevant columns to ensure consistent length
+            cols_needed = [treatment_col] + covariate_cols
```

Ruff (0.14.10)

Consider `[treatment_col, *covariate_cols]` instead of concatenation

Replace with `[treatment_col, *covariate_cols]`

(RUF005)